### PR TITLE
testsuite: enable valgrind testing

### PR DESCRIPTION
**Pull request by garlick**
Just some small updates to get valgrind running again in the test suite.

### CODE REVIEW

File: t/valgrind/valgrind.supp [hunk 1]
```diff
@@ -80,3 +80,13 @@
...
+{
+   <issue_2223>
+   Memcheck:Leak
+   match-leak-kinds: definite
+   fun:malloc
+   fun:dl_open_worker
+   fun:_dl_catch_*
+   fun:_dl_open
+   ...
+}
```
Comment (dongahn) [inline t/valgrind/valgrind.supp:92]: Can we add a new line at this end? The GitHub render marks it. 

It is not an error at all but at the same time we won't need to get this flagged by GitHub either :-)

File: t/valgrind/valgrind-workload.sh [hunk 2]
```diff
@@ -1,13 +1,13 @@
 #!/bin/bash
 
 echo FLUX_URI=$FLUX_URI
-TESTDIR=${SHARNESS_TEST_SRCDIR:-.}/valgrind/workload.d/*
 exitcode=0
 
-# Check for no workload:
-test -d ${TESTDIR} || exit $exitcode
+flux module remove -f sched-simple
+flux module load sched-fluxion-resource
```
Comment (dongahn) [inline t/valgrind/valgrind-workload.sh:7]: I think we need to `load-allowlist` to produce a resource graph conforming to Jobspect V1. 

```suggestion
flux module load sched-fluxion-resource
flux module load sched-fluxion-resource load-allowlist=node,core,gpu
```

File: t/valgrind/valgrind.supp (hunk 1 above)
Comment (garlick) [reply t/valgrind/valgrind.supp:92]: Interesting, I guess the one in core doesn't have one either.  OK I'll add one :-)

**Comment by garlick** [comment]
I went ahead and forced a push with those changes.  Thanks @dongahn.

**Comment by garlick** [comment]
One more force push to fix the issue reference (should have been #506)

**Pull request merged**